src/commands/apply-chain-updates.ts

```diff
@@ -0,0 +1,372 @@
+import { Contract, Interface } from 'ethers'
+import { readFile } from 'fs/promises'
+import { createInterface } from 'readline/promises'
+import bs58 from 'bs58'
+import TokenPoolABI from '../abi/TokenPool.ts'
+import { bigIntReplacer, chainIdFromName, getProviderNetwork } from '../lib/utils.ts'
+import type { Providers } from '../providers.ts'
+import { Format } from './types.ts'
+import { getWallet } from './utils.ts'
+import type { TypedContract } from 'ethers-abitype'
+
+type ChainToAdd = {
+  remoteChainSelector: bigint
+  remotePoolAddresses: string[]
+  remoteTokenAddress: string
+  outboundRateLimiterConfig: {
+    isEnabled: boolean
+    capacity: bigint
+    rate: bigint
+  }
+  inboundRateLimiterConfig: {
+    isEnabled: boolean
+    capacity: bigint
+    rate: bigint
+  }
+}
+
+/**
+ * @description Checks if a given string is in a valid Solana address Base58 format.
+ *
+ * Solana addresses are 32-byte arrays encoded with the Bitcoin Base58 alphabet, resulting
+ * in ASCII text strings of 32-44 characters. This function only validates the character
+ * pattern and length.
+ *
+ * @warning Since this function depends on user's input only, it:
+ * - DOES NOT validate if the address is a valid ed25519 public key
+ * - DOES NOT validate checksum (typos cannot be detected)
+ * - Single character typos, reversed characters, and case errors may still pass
+ *
+ * @param address - A string representing a Solana address in Base58 format.
+ * @returns {boolean} - Returns true if the address is valid, false otherwise.
+ */
+function isSolanaAddress(address: string): boolean {
+  const SOLANA_ADDRESS_REGEX = /^[1-9A-HJ-NP-Za-km-z]{32,44}$/
+
+  return SOLANA_ADDRESS_REGEX.test(address)
+}
+
+/**
+ * @description Converts a Solana address from Base58 format to a 32-byte hex string which CCIP expects.
+ *
+ * If the address is not valid or cannot be decoded to 32 bytes, it returns the original address.
+ *
+ * @param address - A string representing a Solana address in Base58 format.
+ * @returns {string} - Returns the address as a 32-byte hex string prefixed with '0x'.
+ */
+function convertSolanaAddressToBytes32Hex(address: string): string {
+  try {
+    const decoded = bs58.decode(address)
+    if (decoded.length !== 32) {
+      console.warn(
```

Comment:
Can we do the decoded length check in `isSolanaAddress`?  Ideally all validations should be in a single function that returns a boolean.  The convert function should only handle the conversion. That way the convert function is not returning an invalid address (which it does presently when len != 32).  Also the isSolanaAddress function could throw an error as (IIUC) the program should terminate with an invalid address.